contrib/k8s.io/gateway-api/request-mirror.go

```diff
@@ -0,0 +1,182 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+package gatewayapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"mime"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/DataDog/dd-trace-go/v2/appsec"
+	"github.com/DataDog/dd-trace-go/v2/ddtrace/ext"
+	"github.com/DataDog/dd-trace-go/v2/ddtrace/tracer"
+	"github.com/DataDog/dd-trace-go/v2/instrumentation"
+	"github.com/DataDog/dd-trace-go/v2/instrumentation/appsec/dyngo"
+	"github.com/DataDog/dd-trace-go/v2/instrumentation/appsec/emitter/httpsec"
+	"github.com/DataDog/dd-trace-go/v2/instrumentation/httptrace"
+)
+
+const (
+	maxBodyBytes = 5 * 1 << 20 // 5 MB
+)
+
+var (
+	instr  = instrumentation.Load(instrumentation.PackageK8SGatewayAPI)
+	logger = instr.Logger()
+
+	firstRequest sync.Once
+)
+
+// Config holds the configuration for the request mirror server.
+type Config struct {
+	httptrace.ServeConfig
+}
+
+var requestsCount atomic.Uint32
+
+// HTTPRequestMirrorHandler is the handler for the request mirror server.
+// It is made to receive requests from proxies supporting the request mirror feature from the k8s gateway API specification.
+func HTTPRequestMirrorHandler(config Config) http.Handler {
+	if config.ServeConfig.SpanOpts == nil {
+		config.ServeConfig.SpanOpts = []tracer.StartSpanOption{
+			tracer.Tag(ext.SpanKind, ext.SpanKindServer),
+		}
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Force the connection to be closed so we don't send a response
+		wr, ok := w.(http.Hijacker)
+		if !ok {
+			logger.Error("ResponseWriter does not support Hijack")
+			os.Exit(1)
```

Comment:
it feels odd to me to do `os.Exit` here. Maybe a `panic("ResponseWriter does not support Hijack")` could be a more readable way for the users?